### PR TITLE
LLVM D compiler

### PR DESCRIPTION
**Pull request by brainstorm**
As advised by @bgruening on https://github.com/bioconda/bioconda-recipes/pull/8787.

### CODE REVIEW

**Comment by brainstorm** [comment]
I first went for released binaries because I wanted to bypass the complexity required to bootstrap and compile the D compiler as stated here:

https://github.com/ldc-developers/ldc/blob/master/.circleci/config.yml

What would be the best way to tackle this, @isuruf? (which @-handle is there for conda-build-system?)

File: recipes/ldc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  url: https://github.com/ldc-developers/ldc/releases/download/v{{ version }}/ldc2-{{ version }}-windows-x64.7z  # [win]
+  sha256: 17fb0f2f1865e796f05d74fde6b3353c4c1d8a779e2dc51449b9c4b5b5c5ea9a  # [win]
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+    build:
+        - gcc
```
Comment (isuruf) [inline recipes/ldc/meta.yaml]: See https://conda-forge.org/docs/meta.html#compilers

**Comment by brainstorm** [comment]
I guess that the GLIBC error on CircleCI (not failing on TravisCI) might be related with how LDC is linked in the binary releases?: https://github.com/ldc-developers/ldc/releases

I hope I don't have all the way down to bootstrapping the D compiler and so on?:

https://github.com/ldc-developers/ldc/blob/master/.circleci/config.yml

 :_/

**Comment by isuruf** [comment]
You'll have to build from source. Errors are due to the binary being compiled for newer GLIBC version than CentOS6. Here's a build script https://github.com/lomereiter/centos-ldc

**Comment by isuruf** [comment]
Here's what you should do.
1. Package ldc-0.17.1 as in https://github.com/lomereiter/centos-ldc/blob/master/install-software.sh
It's an ancient version, but can be compiled without another ldc compiler.
2. Once ldc-0.17.1 is packaged, use it to compile the latest version with llvmdev from conda-forge. (ldc versioned LLVM has a few patches for android support that we don't need.

**Comment by isuruf** [comment]
Hmm, that was surprisingly easy to build.

**Comment by isuruf** [comment]
@brainstorm, do you mind adding a windows build script?

**Comment by brainstorm** [comment]
Sorry but I don't own any windows machine and I'm quite clueless about it. My latest efforts trying to build ansible for windows failed miserably and since I didn't have the means to debug it further I dropped it, unfortunately :_(

**Comment by isuruf** [comment]
Let's just use the binaries for windows and build from source for unix.

**Comment by isuruf** [comment]
I've been working on it today. Do you mind if I force push?

**Comment by brainstorm** [comment]
Please go ahead, my changes can be discarded ;)

**Comment by brainstorm** [comment]
Awesome work @isuruf, thanks so much for all this! Looks green and fine on all envs, eager to see it in feedstock/upload to conda now 🎉

**Comment by brainstorm** [comment]
@isuruf How come the unix and osx builds are `disabled` in the feedstock? Looks like only the windows build made it through? Or is it just a CI queue issue and those will be uploaded soon?

**Pull request merged**